salt/modules/aptpkg.py

```diff
@@ -2305,6 +2308,16 @@ def mod_repo(repo, saltenv='base', **kwargs):
                 'Error: failed to add key from {0}'.format(key_url)
             )
 
+    elif 'key_text' in kwargs:
+        key_text = kwargs['key_text']
+        cmd = ['apt-key', 'add', '-']
+        out = __salt__['cmd.run_stdout'](cmd, stdin=key_text,
```

Comment:
:+1: for using ``stdin`` and not cat + pipe.

Are we certain that this works with ``python_shell=False``? That is, has importing a key in this fashion been tested? I don't see any integration test here. The reason I ask is that I know when we use ``python_shell=False`` we are telling ``subprocess.Popen()`` not to spawn a shell to run the command (i.e. passing ``shell=False`` to it), and thus certain shell-isms are not available (pipes, and/or operators, etc.). I'm like 90% sure that the ``-`` after the ``apt-key add`` is technically considered an argument to ``apt-key``, and is not the sort of shell-ism that would be disabled with ``python_shell=False``, but since I'm not 100% sure I'm just being cautious here and asking the question.